src/Pact/Eval.hs

```diff
@@ -705,7 +775,35 @@ reduceApp (App (TDef d@Def{..} _) as ai) = {- eperf (asString _dDefName) $ -} do
       Defcap ->
         evalError ai "Cannot directly evaluate defcap"
 reduceApp (App (TLitString errMsg) _ i) = evalError i $ pretty errMsg
-reduceApp (App r _ ai) = evalError ai $ "Expected def: " <> pretty r
+-- show m resolve the function right then and there. make a qualified name, call resolveref. you need to call the TDef case
+reduceApp (App (TDynamic tref tmem _) as ai) = do
+  ref <- reduce tref >>= \case
```

Comment:
Hmm, why does `tref` need to be reduced? I would think it's already been resolved at this point so reduction doesn't buy you anything. Also this will be a `TModRef` right?